File: lms-frontend/src/employee/services/departmentService.js
// departmentService.js

import api from "../../api/axios";

const BASE_URL = "/api/departments";

const getAllDepartments = () => {

    return api.get(
        BASE_URL
    );
};

const getDepartmentById = (departmentId) => {

    return api.get(
        `${BASE_URL}/${departmentId}`
    );
};

const createDepartment = (departmentData) => {

    return api.post(
        BASE_URL,
        departmentData
    );
};

const updateDepartment = (
    departmentId,
    departmentData
) => {

    return api.put(
        `${BASE_URL}/${departmentId}`,
        departmentData
    );
};

const deleteDepartment = (departmentId) => {

    return api.delete(
        `${BASE_URL}/${departmentId}`
    );
};

const departmentService = {

    getAllDepartments,
    getDepartmentById,
    createDepartment,
    updateDepartment,
    deleteDepartment
};

export default departmentService;

File: backend/src/main/java/com/lms_erp/employee/service/DepartmentService.java
package com.lms_erp.employee.service;

import com.lms_erp.employee.dto.DepartmentRequest;
import com.lms_erp.employee.dto.DepartmentResponse;

import java.util.List;

public interface DepartmentService {

    DepartmentResponse createDepartment(DepartmentRequest request);

    DepartmentResponse updateDepartment(Long departmentId,
                                        DepartmentRequest request);

    DepartmentResponse getDepartmentById(Long departmentId);

    List<DepartmentResponse> getAllDepartments();

    void deleteDepartment(Long departmentId);
}